src/transactions/TransactionFrame.h

```diff
@@ -116,28 +106,30 @@ class TransactionFrame : public TransactionFrameBase
                                bool chargeFee,
                                uint64_t lowerBoundCloseTimeOffset,
                                uint64_t upperBoundCloseTimeOffset,
-                               std::optional<FeePair> sorobanResourceFee);
-
-    virtual std::shared_ptr<OperationFrame>
-    makeOperation(Operation const& op, OperationResult& res, size_t index);
+                               std::optional<FeePair> sorobanResourceFee,
+                               TransactionResultPayloadPtr resPayload) const;
 
-    void removeOneTimeSignerFromAllSourceAccounts(AbstractLedgerTxn& ltx) const;
+    void removeOneTimeSignerFromAllSourceAccounts(
+        AbstractLedgerTxn& ltx, TransactionResultPayload& resPayload) const;
 
     void removeAccountSigner(AbstractLedgerTxn& ltxOuter,
                              AccountID const& accountID,
                              SignerKey const& signerKey) const;
 
-    void markResultFailed();
+    void markResultFailed(TransactionResultPayload& resPayload) const;
```

Comment:
I don't think `markFailed` really makes sense since it's so trivial anyway, so I think I'll remove it. I'm a little hesitant about `refundSorobanFee` and `processSeqNum` though. For example, `refundSorobanFee` updates the TxResult payload `feeCharged` (which is definitely in the domain of `TransactionResultPayload`), but it also updates the ledger header fee pool. This feels like a pretty heavyweight operation for the result payload on it's own, and I think I'd like to keep that within the context of a transaction frame instead of just the result payload. I envision `TransactionResultPayload` as a lightweight wrapper around state with helpful getters and setters, and this seems above it's pay grade.

Reply:
Getting rid of `markFailed` SGTM.

`processSeqNum` doesn't need the whole result at all, just the buggy cache (see the comment re cache).

As for `refundSorobanFee`, I agree that moving the whole method to result is probably too much. However, I think it would be beneficial to move specifically handling of the refund in the result to a virtual method overloaded separately for the fee bump result. I'm pretty sure that allows us to get rid of an ugly downcast in `FeeBumpTransactionFrame::processPostApply` and will make code less error-prone going forward (well, I don't anticipate for it to be changed, but at least it will be much easier to read). In general, the more we can avoid using `getInnerResultPayload` in fee bump frame, the better (IDK if we can get rid of all of them, but we should at least try).